<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.5.57">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">


<title>6&nbsp; Data Analysis – Biodiverse Quick Start Guide</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
</style>


<script src="site_libs/quarto-nav/quarto-nav.js"></script>
<script src="site_libs/quarto-nav/headroom.min.js"></script>
<script src="site_libs/clipboard/clipboard.min.js"></script>
<script src="site_libs/quarto-search/autocomplete.umd.js"></script>
<script src="site_libs/quarto-search/fuse.min.js"></script>
<script src="site_libs/quarto-search/quarto-search.js"></script>
<meta name="quarto:offset" content="./">
<link href="./6-export.html" rel="next">
<link href="./4-visualise.html" rel="prev">
<script src="site_libs/quarto-html/quarto.js"></script>
<script src="site_libs/quarto-html/popper.min.js"></script>
<script src="site_libs/quarto-html/tippy.umd.min.js"></script>
<script src="site_libs/quarto-html/anchor.min.js"></script>
<link href="site_libs/quarto-html/tippy.css" rel="stylesheet">
<link href="site_libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="site_libs/bootstrap/bootstrap.min.js"></script>
<link href="site_libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="site_libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script id="quarto-search-options" type="application/json">{
  "location": "sidebar",
  "copy-button": false,
  "collapse-after": 3,
  "panel-placement": "start",
  "type": "textbox",
  "limit": 50,
  "keyboard-shortcut": [
    "f",
    "/",
    "s"
  ],
  "show-item-context": false,
  "language": {
    "search-no-results-text": "No results",
    "search-matching-documents-text": "matching documents",
    "search-copy-link-title": "Copy link to search",
    "search-hide-matches-text": "Hide additional matches",
    "search-more-match-text": "more match in this document",
    "search-more-matches-text": "more matches in this document",
    "search-clear-button-title": "Clear",
    "search-text-placeholder": "",
    "search-detached-cancel-button-title": "Cancel",
    "search-submit-button-title": "Submit",
    "search-label": "Search"
  }
}</script>


</head>

<body class="nav-sidebar floating">

<div id="quarto-search-results"></div>
  <header id="quarto-header" class="headroom fixed-top">
  <nav class="quarto-secondary-nav">
    <div class="container-fluid d-flex">
      <button type="button" class="quarto-btn-toggle btn" data-bs-toggle="collapse" role="button" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">
        <i class="bi bi-layout-text-sidebar-reverse"></i>
      </button>
        <nav class="quarto-page-breadcrumbs" aria-label="breadcrumb"><ol class="breadcrumb"><li class="breadcrumb-item"><a href="./5-data-analysis.html"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Data Analysis</span></a></li></ol></nav>
        <a class="flex-grow-1" role="navigation" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item" aria-controls="quarto-sidebar" aria-expanded="false" aria-label="Toggle sidebar navigation" onclick="if (window.quartoToggleHeadroom) { window.quartoToggleHeadroom(); }">      
        </a>
      <button type="button" class="btn quarto-search-button" aria-label="Search" onclick="window.quartoOpenSearch();">
        <i class="bi bi-search"></i>
      </button>
    </div>
  </nav>
</header>
<!-- content -->
<div id="quarto-content" class="quarto-container page-columns page-rows-contents page-layout-article">
<!-- sidebar -->
  <nav id="quarto-sidebar" class="sidebar collapse collapse-horizontal quarto-sidebar-collapse-item sidebar-navigation floating overflow-auto">
    <div class="pt-lg-2 mt-2 text-left sidebar-header">
    <div class="sidebar-title mb-0 py-0">
      <a href="./">Biodiverse Quick Start Guide</a> 
        <div class="sidebar-tools-main">
    <a href="./Biodiverse-Quick-Start-Guide.pdf" title="Download PDF" class="quarto-navigation-tool px-1" aria-label="Download PDF"><i class="bi bi-file-pdf"></i></a>
</div>
    </div>
      </div>
        <div class="mt-2 flex-shrink-0 align-items-center">
        <div class="sidebar-search">
        <div id="quarto-search" class="" title="Search"></div>
        </div>
        </div>
    <div class="sidebar-menu-container"> 
    <ul class="list-unstyled mt-1">
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./index.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">1</span>&nbsp; <span class="chapter-title"><strong>Quick Start Guide</strong></span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./1-overview.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">2</span>&nbsp; <span class="chapter-title">Overview</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./2-starting.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">3</span>&nbsp; <span class="chapter-title">Starting Biodiverse</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./3-import.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">4</span>&nbsp; <span class="chapter-title">Importing Data</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./4-visualise.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Visualising Data</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./5-data-analysis.html" class="sidebar-item-text sidebar-link active">
 <span class="menu-text"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Data Analysis</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./6-export.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Export</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./7-summary.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">8</span>&nbsp; <span class="chapter-title">Summary</span></span></a>
  </div>
</li>
        <li class="sidebar-item">
  <div class="sidebar-item-container"> 
  <a href="./8-acknowledgements.html" class="sidebar-item-text sidebar-link">
 <span class="menu-text"><span class="chapter-number">9</span>&nbsp; <span class="chapter-title">Acknowledgements</span></span></a>
  </div>
</li>
    </ul>
    </div>
</nav>
<div id="quarto-sidebar-glass" class="quarto-sidebar-collapse-item" data-bs-toggle="collapse" data-bs-target=".quarto-sidebar-collapse-item"></div>
<!-- margin-sidebar -->
    <div id="quarto-margin-sidebar" class="sidebar margin-sidebar">
        <nav id="TOC" role="doc-toc" class="toc-active">
    <h2 id="toc-title">Table of contents</h2>
   
  <ul>
  <li><a href="#5.1-building-a-spatial-index" id="toc-5.1-building-a-spatial-index" class="nav-link active" data-scroll-target="#5.1-building-a-spatial-index"><span class="header-section-number">6.1</span> Building a Spatial Index</a></li>
  <li><a href="#5.2-running-a-cluster-analysis" id="toc-5.2-running-a-cluster-analysis" class="nav-link" data-scroll-target="#5.2-running-a-cluster-analysis"><span class="header-section-number">6.2</span> Running a Cluster Analysis**</a>
  <ul class="collapse">
  <li><a href="#5.2.1-setting-cluster-analysis-parameters" id="toc-5.2.1-setting-cluster-analysis-parameters" class="nav-link" data-scroll-target="#5.2.1-setting-cluster-analysis-parameters"><span class="header-section-number">6.2.1</span> Setting cluster analysis parameters**</a></li>
  <li><a href="#viewing-the-cluster-results" id="toc-viewing-the-cluster-results" class="nav-link" data-scroll-target="#viewing-the-cluster-results"><span class="header-section-number">6.2.2</span> Viewing the cluster results</a></li>
  </ul></li>
  <li><a href="#running-a-spatial-moving-window-analysis-5.3-running-a-spatial-moving-window-analysis" id="toc-running-a-spatial-moving-window-analysis-5.3-running-a-spatial-moving-window-analysis" class="nav-link" data-scroll-target="#running-a-spatial-moving-window-analysis-5.3-running-a-spatial-moving-window-analysis"><span class="header-section-number">6.3</span> Running a Spatial (Moving Window) Analysis** {#5.3-running-a-spatial-(moving-window)-analysis}</a>
  <ul class="collapse">
  <li><a href="#introduction" id="toc-introduction" class="nav-link" data-scroll-target="#introduction"><span class="header-section-number">6.3.1</span> Introduction</a></li>
  <li><a href="#setting-the-spatial-analysis-options" id="toc-setting-the-spatial-analysis-options" class="nav-link" data-scroll-target="#setting-the-spatial-analysis-options"><span class="header-section-number">6.3.2</span> Setting the Spatial Analysis Options</a></li>
  <li><a href="#selecting-indices-to-calculate" id="toc-selecting-indices-to-calculate" class="nav-link" data-scroll-target="#selecting-indices-to-calculate"><span class="header-section-number">6.3.3</span> Selecting indices to calculate</a></li>
  <li><a href="#viewing-the-spatial-analysis-results" id="toc-viewing-the-spatial-analysis-results" class="nav-link" data-scroll-target="#viewing-the-spatial-analysis-results"><span class="header-section-number">6.3.4</span> Viewing the Spatial Analysis Results</a></li>
  </ul></li>
  <li><a href="#section" id="toc-section" class="nav-link" data-scroll-target="#section"><span class="header-section-number">6.4</span> </a>
  <ul class="collapse">
  <li><a href="#running-analysis-with-local-neighbours" id="toc-running-analysis-with-local-neighbours" class="nav-link" data-scroll-target="#running-analysis-with-local-neighbours"><span class="header-section-number">6.4.1</span> Running analysis with local neighbours</a></li>
  </ul></li>
  <li><a href="#running-a-randomisation-analysis" id="toc-running-a-randomisation-analysis" class="nav-link" data-scroll-target="#running-a-randomisation-analysis"><span class="header-section-number">6.5</span> Running a Randomisation Analysis</a>
  <ul class="collapse">
  <li><a href="#introduction-1" id="toc-introduction-1" class="nav-link" data-scroll-target="#introduction-1"><span class="header-section-number">6.5.1</span> Introduction</a></li>
  <li><a href="#spatially-structured-randomisations" id="toc-spatially-structured-randomisations" class="nav-link" data-scroll-target="#spatially-structured-randomisations"><span class="header-section-number">6.5.2</span> Spatially Structured Randomisations</a></li>
  </ul></li>
  </ul>
</nav>
    </div>
<!-- main -->
<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title"><span class="chapter-number">6</span>&nbsp; <span class="chapter-title">Data Analysis</span></h1>
</div>



<div class="quarto-title-meta">

    
  
    
  </div>
  


</header>


<section id="5.1-building-a-spatial-index" class="level2" data-number="6.1">
<h2 data-number="6.1" class="anchored" data-anchor-id="5.1-building-a-spatial-index"><span class="header-section-number">6.1</span> Building a Spatial Index</h2>
<p>We will build a <strong>spatial index</strong> for our new BaseData object. A spatial index enables analyses using this BaseData to run faster (though beware that some user-defined <a href="#5.3.2-setting-the-spatial-analysis-options">spatial conditions</a> will not work with a spatial index). Note that an index can be deleted/rebuilt at any time, and these changes affect only subsequent analyses, not those that preceded any change.</p>
<p>Select <em>Build Spatial Index</em> from the <em>Analyses</em> menu option:</p>
<p>![][image29]</p>
<p>For most purposes, accept the index sizes suggested by the system (see <a href="https://github.com/shawnlaffan/biodiverse/wiki/KeyConcepts#using-the-spatial-index">Using the Spatial Index</a> for additional details on setting optimal index sizes).</p>
<p>Click <strong><em>OK</em></strong> to build the spatial index. A pop-up dialog notifies you that the index has been built (<strong><em>OK</em></strong>).</p>
<p>![][image30]</p>
</section>
<section id="5.2-running-a-cluster-analysis" class="level2" data-number="6.2">
<h2 data-number="6.2" class="anchored" data-anchor-id="5.2-running-a-cluster-analysis"><span class="header-section-number">6.2</span> Running a Cluster Analysis**</h2>
<p>Once you have imported or opened a BaseData object, you can run cluster analyses to identify clusters in this data.</p>
<p>Biodiverse supports agglomerative clustering of the groups based on their labels, or some function of their labels such as the values of a linked matrix or tree.</p>
<p>We will run a cluster analysis on the BaseData imported earlier ‘<em>Example_new_basedata’</em> (see <a href="#3.3-importing-data">Importing data</a>). If <em>My_new_basedata</em> is not loaded, but you have saved it, open this file now by clicking on ‘<em>Open</em>’ from the <em>Basedata</em> menu and navigating to the location of your bds file.</p>
<p>![][image31]</p>
<p>To run a cluster analysis on the currently selected BaseData object, select <em>Cluster</em> from the <em>Analyses</em> menu:![][image32]</p>
<p>The cluster analysis tab opens. It has three main sections where you can set options:</p>
<p>![][image33]</p>
<ul>
<li>The upper section (<em>Parameters</em>) determines the parameters used in the clustering to generate a tree. This includes options to control the cluster tie-breaker algorithm.<br>
</li>
<li>There are check-boxes in the middle section that can be toggled to alter how the cluster analysis runs, e.g.&nbsp;by controlling aspects like process performance (e.g.&nbsp;memory usage), the automatic export of results, etc.<br>
</li>
<li>The lower (<em>Spatial calculations to run for each cluster node</em>) determines what subsequent calculations will be run for each node in this tree, using the groups it contains to define the spatial sample. This allows you to, for example, calculate the phylogenetic diversity of the regions defined by each cluster node.</li>
</ul>
<section id="5.2.1-setting-cluster-analysis-parameters" class="level3" data-number="6.2.1">
<h3 data-number="6.2.1" class="anchored" data-anchor-id="5.2.1-setting-cluster-analysis-parameters"><span class="header-section-number">6.2.1</span> Setting cluster analysis parameters**</h3>
<p>For the purposes of this guide, accept the default <strong>name</strong> for the cluster analyses (see diagram above) and also the default <strong>metric</strong> ‘<em>SORENSON</em>’ (the set of indices that can be used as the clustering metric is denoted by a column in the <a href="https://github.com/shawnlaffan/biodiverse/wiki/Indices">Indices</a> list). We will also accept the default <strong>linkage</strong> of ‘<em>link_average</em>’, which calculates the dissimilarity of each newly created group with all other groups as the average value of the groups merged to form it, weighted by the number of terminal nodes they each include. In this way a merger between node A with 10 terminal nodes and node B with 1 terminal node will not be biased towards the labels in node B (see the <a href="https://github.com/shawnlaffan/biodiverse/wiki/SampleSession#running-a-cluster-analysis">Cluster parameters</a> section of the Sample Session for details of the additional linkage functions that are available).</p>
<p>The <strong>spatial conditions</strong> can be used to control which Groups are considered candidates to be clustered together. The default condition in the ‘<em>Spatial conditions set 1</em>’ text box provided by the system is ‘<em>sp_select_all()</em>’. This condition selects every group and is an appropriate condition to use for unconstrained clustering. If you wish to use a different spatial condition, for example to first cluster within geographic regions, then refer to <a href="https://github.com/shawnlaffan/biodiverse/wiki/SpatialConditions">Spatial Conditions</a> for further details. The <strong>definition query</strong> serves a similar purpose, except that it can be used to restrict the clustering to only a subset of groups. For example you might have a data set for Europe but only want to cluster cells in regions above a specified elevation value. The <a href="#5.3-running-a-spatial-(moving-window)-analysis">Spatial Analysis</a> section in this document has some more details about spatial conditions and definition queries.</p>
<p>Check the <strong>option</strong> or <strong>calculations</strong> boxes as appropriate. If you’re curious as to what they do, hover over them with the mouse pointer to view a tooltip.</p>
<p>Click on the <strong><em>Go!</em></strong> button (keyboard shortcut <em>Control-G</em>). The system will first build the dissimilarity matrix (or matrices if you have several spatial conditions), then run the clustering using these matrices, and then any spatial calculations you have selected for each node. It will then prompt you to <strong><em>display results</em></strong>.</p>
</section>
<section id="viewing-the-cluster-results" class="level3" data-number="6.2.2">
<h3 data-number="6.2.2" class="anchored" data-anchor-id="viewing-the-cluster-results"><span class="header-section-number">6.2.2</span> Viewing the cluster results</h3>
<p>There are three sub-panes within the display pane. On the left is the group grid, on the upper right is the tree representing the clustering, and on the lower right is the scree plot for the tree.</p>
<p>As with the visualisation of basedata/matrix/tree objects described earlier, the system is linked, and interactions in the group grid or tree of the cluster display are generally reflected in the other. Hovering the mouse over a node in the tree highlights the groups (in the group grid) that are contained in that node. Likewise, hovering the mouse over a group in the group grid will highlight the path (set of nodes/clusters) in the tree to which that group belongs.</p>
<p>Left click on a tree node to colour a set of subclusters (descendant nodes in the tree). These clusters are split into a number of coloured groupings based on the “<em>Clusters to colour</em>” parameter at the bottom of the pane. Note that some leaf nodes in the tree may have a length of zero (indicated by vertical bars at the leftmost side of the tree, longer vertical bars indicating more zero-length leaf nodes). Thus, if any such nodes occur under (to the left of) the node you have selected, their colouring will not be apparent if you are using “<em>Plot by Length</em>” mode. These nodes, along with their colouring (if selected) can be made apparent by switching to “<em>Plot by Depth</em>” mode under the tree options button under the <em>Display</em> menu at the left.</p>
<p>The blue sliding bar in the tree pane can be dragged across the tree to colour the nodes and groups at that level. The bar also displays the number of nodes it is crossing when the mouse is focused on it. The maximum number of contrasting colours the system will display is 13. If the slider bar crosses more than 13 nodes, all nodes will be uniformly coloured red instead, and groups in the group grid will not be coloured.</p>
<p>It is also possible to <a href="https://biodiverse-analysis-software.blogspot.com/2016/09/new-selection-tool-in-cluster-analysis.html">assign your own colours to the clusters</a> and to generate a geotiff (raster) to <a href="https://biodiverse-analysis-software.blogspot.com/2018/08/cluster-analyses-export-coloured.html">reproduce the coloured cells in other systems</a>.</p>
</section>
</section>
<section id="running-a-spatial-moving-window-analysis-5.3-running-a-spatial-moving-window-analysis" class="level2" data-number="6.3">
<h2 data-number="6.3" class="anchored" data-anchor-id="running-a-spatial-moving-window-analysis-5.3-running-a-spatial-moving-window-analysis"><span class="header-section-number">6.3</span> Running a Spatial (Moving Window) Analysis** {#5.3-running-a-spatial-(moving-window)-analysis}</h2>
<section id="introduction" class="level3" data-number="6.3.1">
<h3 data-number="6.3.1" class="anchored" data-anchor-id="introduction"><span class="header-section-number">6.3.1</span> Introduction</h3>
<p>The Spatial Analyses are a type of moving window analyses (because nearly all analyses in Biodiverse are spatial in some way), and are another method of identifying spatial patterns in your data. Depending upon the exact nature of your data, it is likely that the Groups comprising your BaseData constitute a geographic surface. Moving windows are used to systematically iterate across this space, with one or two <strong>neighbourhoods</strong> of the moving window being used to assess the labels/values aggregated in the Groups. The window might also consist only of the processing group, with no neighbours (each group is analysed separately).</p>
<p>To run a moving window analysis, you will need to have <a href="#open-or-import-data">imported or loaded</a> a BaseData object and ideally built a <a href="#heading=h.3as4poj">spatial index</a> for this BaseData. These instructions assume you have either imported the sample data above, or you can open the example BaseData object ’<em>example_data.bds</em>’in the Biodiverse data folder.</p>
<p>Select a Basedata object by clicking on it and open the menu option <em>Analyses -&gt; Spatial</em> and the spatial tab appears:</p>
<p>![][image34]</p>
<p>The Spatial tab has two main sections where you can set options. The upper section (<em>Parameters</em>) determines the parameters used in defining the two neighbour sets used in the spatial analysis (the second is optional) as well as the definition query. The lower (<em>Calculations to run for each neighbourhood</em>) determines the subsequent calculations that will be run for the set of neighbours related to each group. You can select any number of calculations to perform.</p>
</section>
<section id="setting-the-spatial-analysis-options" class="level3" data-number="6.3.2">
<h3 data-number="6.3.2" class="anchored" data-anchor-id="setting-the-spatial-analysis-options"><span class="header-section-number">6.3.2</span> Setting the Spatial Analysis Options</h3>
<p>The <strong>Name</strong> option is the name used in the system. You can accept the default, or enter a new name (you cannot have duplicate names within the same BaseData object).</p>
<p>The <strong>Neighbour set 1</strong> and <strong>Neighbour set 2</strong> text boxes allow you to define the neighbour sets used for the calculations (see <a href="https://github.com/shawnlaffan/biodiverse/wiki/SpatialConditions">Spatial Conditions</a> for further details). The size and shape of the neighbourhoods defines the set of groups to be considered in an analysis and it is up to you to decide what sort of neighbourhoods and how many to use for your analysis. The system provides considerable flexibility in this regard: both neighbour sets may be arbitrarily defined independently of each other and you are not obliged to specify a second neighbourhood.</p>
<p>The default condition in the <em>Neighbour set</em> 1 text box is ‘<em>sp_self_only()</em>’. This restricts the neighbourhood to one cell/group (the processing group, which is the group being processed at an iteration). The default condition in the <em>Neighbour set 2</em> text box ‘<em>sp_circle (radius =&gt; 100000)</em>’ defines a circular neighbourhood of 100000 units centred on the processing group. The units are whatever the original data used, and for the example data this is a 100 km or one cell radius.</p>
<p>These default conditions are appropriate for our sample data, but we will modify the second condition. Change the circle radius from <em>100000</em> to <em>200000</em>.</p>
<p>See <a href="https://github.com/shawnlaffan/biodiverse/wiki/SpatialConditions">Spatial Conditions</a> for more detail on defining the neighbourhood sets (e.g.&nbsp;for the range of functions that you can use to specify neighbourhoods).</p>
<p>Note that the use of neighbourhoods varies according to the type of calculation selected for analysis (see <a href="#5.3.3-selecting-indices-to-calculate">Selecting Indices</a> below). Some calculations aggregate the neighbourhoods into a single set (e.g.&nbsp;<a href="http://code.google.com/p/biodiverse/wiki/Indices#Endemism_and_Rarity">endemism_whole</a>), others compare the labels in the first neighbour set with those in the second (e.g.&nbsp;<a href="http://code.google.com/p/biodiverse/wiki/Indices#Taxonomic_Dissimilarity_and_Comparison">Jaccard and other dissimilarity indices</a>), while others use the first set to define the list of labels to use but then consider distributions across both neighbour sets (e.g.&nbsp;<a href="http://code.google.com/p/biodiverse/wiki/Indices#Endemism_and_Rarity">endemism_central</a>). Some will not be run unless two neighbour sets are defined, primarily the turnover calculations.</p>
<p>The <strong>Definition query</strong> text box allows the analysis to be applied to only a subset of groups (those which satisfy the criteria in the definition query, but note that all groups are still considered as possible neighbours). Leave this blank for this session. For details of setting a definition query see <a href="https://github.com/shawnlaffan/biodiverse/wiki/SpatialConditions#definition-queries">Definition Queries</a>.</p>
<p>The Verify buttons let you know if the spatial conditions entered are syntactically valid.</p>
</section>
<section id="selecting-indices-to-calculate" class="level3" data-number="6.3.3">
<h3 data-number="6.3.3" class="anchored" data-anchor-id="selecting-indices-to-calculate"><span class="header-section-number">6.3.3</span> Selecting indices to calculate</h3>
<p>Use the check boxes to select which calculations are to be calculated. Press the plus buttons next to the <em>Endemism,</em> <em>Lists and Counts</em> and <em>Taxonomic Dissimilarity and Comparison</em> sets to expand their trees. Check the boxes of <em>Endemism central</em>, <em>Redundancy</em> and <em>Richness</em> under <em>Lists and Counts</em> and <em>Sorenson</em> under the <em>Taxonomic Dissimilarity and Comparison</em> set.</p>
<p>Details about the full range of indices are available from the <a href="https://github.com/shawnlaffan/biodiverse/wiki/Indices">Indices</a> page.</p>
<p>Click on the <strong><em>Go!</em></strong> button. The system will then run the selected spatial analyses.</p>
<p>![][image35]</p>
</section>
<section id="viewing-the-spatial-analysis-results" class="level3" data-number="6.3.4">
<h3 data-number="6.3.4" class="anchored" data-anchor-id="viewing-the-spatial-analysis-results"><span class="header-section-number">6.3.4</span> Viewing the Spatial Analysis Results</h3>
<p>Once the analysis is complete, the system asks you whether you want to display the results. Click <em>Yes</em> and you will be shown a map of the moving window analysis results. Pull the pane down to view the options you used, for example to change them and re-run the analysis.</p>
<p>![][image36]</p>
<p>Vector overlays from shapefiles can be plotted using the “overlays” interface at the bottom of the screen. Click the <em>Overlays…</em> button, click <em>Add</em>, navigate to the Biodiverse/data folder and select <em>coastline_lamberts.shp</em>, click <em>OK</em>, highlight this file in the Select Overlay dialog box and click <em>OK</em>.</p>
<p>Nothe that Biodiverse is not yet aware of coordinate systems and only “sees” the coordinates as numbers. This means that the vector data MUST be in the same coordinate system as your basedata. If you have imported data as lat/long coordinates (a geographic coordinate system) then the vector data must also use geographic coordinates. If your basedata are in an equal area coordinate system then your vector data must also be in that coordinate system.</p>
<p>![][image37]</p>
<p>![][image38]</p>
</section>
</section>
<section id="section" class="level2" data-number="6.4">
<h2 data-number="6.4" class="anchored" data-anchor-id="section"><span class="header-section-number">6.4</span> </h2>
<section id="running-analysis-with-local-neighbours" class="level3" data-number="6.4.1">
<h3 data-number="6.4.1" class="anchored" data-anchor-id="running-analysis-with-local-neighbours"><span class="header-section-number">6.4.1</span> Running analysis with local neighbours</h3>
<p>In the above moving window spatial analysis, the output visualisations only provided a single cell output of iterations within the map (black dot within the group cell).</p>
<p>However, altering the second neighbour entry to ”sp_circle (radius = &gt; 500000)” when setting the initial analysis parameters (noted above), can identify aggregated relationships of these cells with their “neighbours”.</p>
<p>![][image39]</p>
<p>The result is a display like this:</p>
<p>![][image40]</p>
<p>Hovering over a group will highlight the groups used in the calculations – a solid circle for those in the first neighbourhood and a dash for those in the second neighbourhood. Right click on a group to keep the highlighting at that group until the mouse is left clicked on any group. You can control whether these are displayed, or just display one set, by using the Neighbours selection options.</p>
<p>Holding the control key while clicking on a group, or clicking on a group with the middle mouse button, produces a pop-up list with all the results in it. It also includes lists of the elements (groups) in each neighbour set (Elements set1, set2 and all) and of the labels in these neighbour sets (Labels set1, set2 and all). The “All” list is the union of neighbour sets 1 and 2.</p>
<p>The tree branches are coloured according to whether they are found only in the first neighbour set (blue), the second neighbour set (red) , both (black) or neither (grey). See also the <a href="https://biodiverse-analysis-software.blogspot.com/2016/01/more-on-tree-visualisations-in.html">blog post</a>. <a href="https://biodiverse-analysis-software.blogspot.com/2017/09/visualise-spatial-analysis-results-on.html">Some indices can also be plotted on the tree</a>.</p>
</section>
</section>
<section id="running-a-randomisation-analysis" class="level2" data-number="6.5">
<h2 data-number="6.5" class="anchored" data-anchor-id="running-a-randomisation-analysis"><span class="header-section-number">6.5</span> Running a Randomisation Analysis</h2>
<section id="introduction-1" class="level3" data-number="6.5.1">
<h3 data-number="6.5.1" class="anchored" data-anchor-id="introduction-1"><span class="header-section-number">6.5.1</span> Introduction</h3>
<p>Randomisations in Biodiverse are used to assess the statistical significance of a set of analysis results given some randomisation scheme such as shuffling the species around the map (labels across groups in Biodiverse terms), subject to constraints such as each cell (group) must maintain the same number of species. Randomisations are key to interpreting where results differ from what would be expected, and are integral to protocols such as the Categorical Analysis of Paleo and Neo Endemism (<a href="https://biodiverse-analysis-software.blogspot.com/2022/10/biodiverse-now-calculates-canape-for-you.html">CANAPE</a>). The output can provide rank-relative scores from which significance can be derived as well as <a href="https://biodiverse-analysis-software.blogspot.com/2020/11/randomisations-now-also-generate-z.html">z-scores</a>.</p>
<p>The default randomisation algorithm in Biodiverse is called <a href="https://biodiverse-analysis-software.blogspot.com/2020/11/randomisations-how-randstructured.html">rand_structured</a>.</p>
<p>The rand_structured algorithm is so called because it maintains the structure of the original data, specifically the spatial distribution of the richness patterns, while randomly allocating taxa (labels) to groups (cells). By default, the richness patterns are matched exactly, but users also have the option to allow tolerances such as an additional number of labels per group (e.g.&nbsp;five extra), or some multiple of the observed (e.g.&nbsp;1.5 times as many). The rand_structured randomisation uses a filling algorithm, followed by a swapping algorithm to reach its richness targets. There is a <a href="https://biodiverse-analysis-software.blogspot.com/search/label/randomisations">series of blog posts</a> describing the randomisation algorithms and visualisations.</p>
<p>The basic process of the randomisation is this. For each iteration of the randomisation analysis, Biodiverse will:</p>
<ol type="1">
<li>Create a new basedata object with a random allocation of labels to groups, according to the selected algorithm<br>
</li>
<li>For each analysis in the basedata:
<ol type="1">
<li>Regenerate a version using the randomised basedata<br>
</li>
<li>Compare the values of the analyses from the original and randomised basedata and track if they are higher or lower, on a group by group basis<br>
</li>
<li>Track basic statistics of the distribution to allow the calculation of the mean and standard deviation, and thus z-scores.</li>
</ol></li>
</ol>
<p>Randomisations can be run by selecting the Analyses&gt;Randomisation menu option:</p>
<p>![][image41]</p>
<p>Cluster and Region Grower trees are not rebuilt by default, partly for speed, partly because it is not normally needed. If you do want to do so then select the “build randomised tree outputs” checkbox. Note that any calculations per node are compared against the randomly generated basedata, though.</p>
<p>![][image42]</p>
<p>Once the relevant parameters are specified, press “<strong><em>Randomise!</em></strong>”.</p>
<p><strong><em>Note that the randomisation analysis tab does not show maps or any other displays once the specified number of iterations have completed</em></strong>. The results are added as lists to any existing analysis objects and can be displayed by selecting from the “List and Indices” option on the relevant plots. You might need to close any open plots and re-open them to see the new lists. See the <a href="https://github.com/shawnlaffan/biodiverse/wiki/AnalysisTypes#where-do-the-randomisation-results-go-and-what-do-they-mean">main help system</a>.</p>
<p>A description of what the list and index names mean is also in the <a href="https://github.com/shawnlaffan/biodiverse/wiki/AnalysisTypes#where-do-the-randomisation-results-go-and-what-do-they-mean">main help system</a>. (Consider for example the Rand1&gt;&gt;SPATIAL_RESULTS list and C_END_CWE index below).</p>
<p>![][image43]</p>
<p>![][image44]</p>
</section>
<section id="spatially-structured-randomisations" class="level3" data-number="6.5.2">
<h3 data-number="6.5.2" class="anchored" data-anchor-id="spatially-structured-randomisations"><span class="header-section-number">6.5.2</span> Spatially Structured Randomisations</h3>
<p>In the standard implementation of a randomisation, labels are randomly allocated to any group across the entire data set. This works well for many studies and is very effective, however, when one begins to scale analyses to larger extents, then the total pool of labels begins to span many different environments. The randomisations could allocate a polar taxon to the tropics, and a desert taxon to a rainforest. This does not make the randomisation invalid, but it is perhaps not as strict as it could be, and perhaps also not as realistic.</p>
<p>This can be readily fixed in Biodiverse by specifying a spatial condition to define subsets. Or, if you only want to randomise a subset of your data while holding the rest as-observed, you can use a <em>definition</em> query. You can specify spatial conditions using the same syntax as for the Spatial and Cluster analyses. The analytical decisions are yours but generally speaking it is better to use a condition that generates non-overlapping regions. Each group is allocated to only one subset, so in such overlapping cases it is the first one that contains a group “wins” that group.</p>
<p>To read more on spatially structured randomisations see <a href="https://biodiverse-analysis-software.blogspot.com/search?q=structured+randomisation">this</a> blog post.</p>
<p>![][image45]</p>


</section>
</section>

</main> <!-- /main -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const onCopySuccess = function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  }
  const getTextToCopy = function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
    text: getTextToCopy
  });
  clipboard.on('success', onCopySuccess);
  if (window.document.getElementById('quarto-embedded-source-code-modal')) {
    // For code content inside modals, clipBoardJS needs to be initialized with a container option
    // TODO: Check when it could be a function (https://github.com/zenorocha/clipboard.js/issues/860)
    const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
      text: getTextToCopy,
      container: window.document.getElementById('quarto-embedded-source-code-modal')
    });
    clipboardModal.on('success', onCopySuccess);
  }
    var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
    var mailtoRegex = new RegExp(/^mailto:/);
      var filterRegex = new RegExp('/' + window.location.host + '/');
    var isInternal = (href) => {
        return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
    }
    // Inspect non-navigation links and adorn them if external
 	var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
    for (var i=0; i<links.length; i++) {
      const link = links[i];
      if (!isInternal(link.href)) {
        // undo the damage that might have been done by quarto-nav.js in the case of
        // links that we want to consider external
        if (link.dataset.originalHref !== undefined) {
          link.href = link.dataset.originalHref;
        }
      }
    }
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      if (note) {
        return note.innerHTML;
      } else {
        return "";
      }
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      // TODO in 1.5, we should make sure this works without a callout special case
      if (note.classList.contains("callout")) {
        return note.outerHTML;
      } else {
        return note.innerHTML;
      }
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
<nav class="page-navigation">
  <div class="nav-page nav-page-previous">
      <a href="./4-visualise.html" class="pagination-link" aria-label="Visualising Data">
        <i class="bi bi-arrow-left-short"></i> <span class="nav-page-text"><span class="chapter-number">5</span>&nbsp; <span class="chapter-title">Visualising Data</span></span>
      </a>          
  </div>
  <div class="nav-page nav-page-next">
      <a href="./6-export.html" class="pagination-link" aria-label="Export">
        <span class="nav-page-text"><span class="chapter-number">7</span>&nbsp; <span class="chapter-title">Export</span></span> <i class="bi bi-arrow-right-short"></i>
      </a>
  </div>
</nav>
</div> <!-- /content -->




</body></html>